### ISSUE TITLE
Undefined method match for nil:NilClass (NoMethodError)

### ISSUE THREAD

**Issue by connorshea** [tjgrathwell/rails5-spec-converter#1]
You can see this error on GitLab under Rails 4.2.7 or Rails 5.0.0.

Unfortunately I'm not sure how to tell what file is causing this since the trace isn't giving me much of anything. It works for the first ~32 specs though!

```
/Users/connor/.rvm/gems/ruby-2.2.4/gems/rails5-spec-converter-1.0.3/lib/rails5/spec_converter/text_transformer.rb:54:in `block in looks_like_route_definition?': undefined method `match' for nil:NilClass (NoMethodError)
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/rails5-spec-converter-1.0.3/lib/rails5/spec_converter/text_transformer.rb:50:in `each'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/rails5-spec-converter-1.0.3/lib/rails5/spec_converter/text_transformer.rb:50:in `looks_like_route_definition?'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/rails5-spec-converter-1.0.3/lib/rails5/spec_converter/text_transformer.rb:31:in `block in transform'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:245:in `block in visit_descendants_with_types'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:243:in `each'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:243:in `visit_descendants_with_types'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:246:in `block in visit_descendants_with_types'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:243:in `each'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:243:in `visit_descendants_with_types'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:246:in `block in visit_descendants_with_types'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:243:in `each'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:243:in `visit_descendants_with_types'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:246:in `block in visit_descendants_with_types'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:243:in `each'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:243:in `visit_descendants_with_types'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:246:in `block in visit_descendants_with_types'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:243:in `each'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:243:in `visit_descendants_with_types'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/astrolabe-1.3.1/lib/astrolabe/node.rb:226:in `each_node'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/rails5-spec-converter-1.0.3/lib/rails5/spec_converter/text_transformer.rb:25:in `transform'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/rails5-spec-converter-1.0.3/lib/rails5/spec_converter/cli.rb:9:in `block in run'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/rails5-spec-converter-1.0.3/lib/rails5/spec_converter/cli.rb:7:in `glob'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/rails5-spec-converter-1.0.3/lib/rails5/spec_converter/cli.rb:7:in `run'
	from /Users/connor/.rvm/gems/ruby-2.2.4/gems/rails5-spec-converter-1.0.3/exe/rails5-spec-converter:6:in `<top (required)>'
	from /Users/connor/.rvm/gems/ruby-2.2.4/bin/rails5-spec-converter:23:in `load'
	from /Users/connor/.rvm/gems/ruby-2.2.4/bin/rails5-spec-converter:23:in `<main>'
	from /Users/connor/.rvm/gems/ruby-2.2.4/bin/ruby_executable_hooks:15:in `eval'
	from /Users/connor/.rvm/gems/ruby-2.2.4/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by connorshea** [comment]
It may be either [`compare_controller_spec.rb`](https://github.com/gitlabhq/gitlabhq/blob/6f8156cea587f08e11ad578bec012af103cc82e4/spec/controllers/projects/compare_controller_spec.rb) or [`commits_controller_spec.rb`](https://github.com/gitlabhq/gitlabhq/blob/6f8156cea587f08e11ad578bec012af103cc82e4/spec/controllers/projects/commits_controller_spec.rb) since the latter is the last file it modifies before crashing.

**Comment by connorshea** [comment]
Maybe related to #2?

**Comment by connorshea** [comment]
I'll do some quick binary searching for the offending test.

**Comment by connorshea** [comment]
In `compare_controller_spec.rb` it seems to fail on either of these tests, but if you leave everything else it works fine:

```ruby
it 'compare should show some diffs' do
    get(:show,
        namespace_id: project.namespace.to_param,
        project_id: project.to_param,
        from: ref_from,
        to: ref_to)

    expect(response).to be_success
    expect(assigns(:diffs).first).not_to be_nil
    expect(assigns(:commits).length).to be >= 1
  end
```

```ruby
it 'invalid source ref' do
      get(:show,
          namespace_id: project.namespace.to_param,
          project_id: project.to_param,
          from: 'non-existent',
          to: ref_to)

      expect(response).to be_success
      expect(assigns(:diffs).to_a).to eq([])
      expect(assigns(:commits)).to eq([])
    end
```

**Comment by tjgrathwell** [comment]
Fixed in 1.0.4. It seems to execute cleanly on https://github.com/gitlabhq/gitlabhq now, though whether the results are correct is anyone's guess 😨

**Comment by connorshea** [comment]
👏  awesome!
